context/src/repo/mod.rs

```diff
@@ -157,6 +165,76 @@ impl BundleRepo {
         }
         Ok(None)
     }
+
+    #[cfg(feature = "git-access")]
+    fn resolve_repo_head_commit<'a>(
+        git_repo: &'a gix::Repository,
+        current_commit: gix::Commit<'a>,
+        repo_head_branch: String,
+    ) -> gix::Commit<'a> {
+        // for GH actions, grab PR branch HEAD commit, not the PR merge commit
+        let re = Regex::new(r"refs\/remotes\/pull\/[0-9]+\/merge").unwrap();
```

Comment:
nit: regexs should be created in a `lazy_static` so we don't try to compile them on every invocation:
https://docs.rs/regex/latest/regex/#avoid-re-compiling-regexes-especially-in-a-loop